com.playeveryware.eos/Runtime/Core/EOS_SDK_Additions/FileRequestTransferWrapper.cs

```diff
@@ -0,0 +1,142 @@
+        /// Wrapped instance.
+        /// </summary>
+        protected T _instance;
+
+        /// <summary>
+        /// Creates a new instance of the FileRequestTransferWrapper.
+        /// </summary>
+        /// <param name="instance">
+        /// The class being wrapped.
+        /// </param>
+        protected FileRequestTransferWrapper(T instance)
+        {
+            _instance = instance;
+        }
+
+        #region Equality operator overloads
+
+        /*
+         * The following section contains methods that are used to seemlessly
+         * provide wrapper functionality for the classes that this class wraps,
+         * allowing the wrapping to be transparent when assigning values to
+         * instances of the wrapper, or when comparing the instance to null (for
+         * instance in scenarios where the instance being wrapped is null, but
+         * the wrapper is not null - in those cases we still want to say it's
+         * equal to null to be as transparent of a wrapper as possible).
+         */
+
+        public static bool operator ==(FileRequestTransferWrapper<T> wrapper, object obj)
+        {
+            if (wrapper is null || wrapper._instance is null)
+            {
+                return obj is null;
+            }
+
+            return wrapper._instance.Equals(obj);
+        }
+
+        public static bool operator !=(FileRequestTransferWrapper<T> wrapper, object obj)
+        {
+            return !(wrapper == obj);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(this, obj)) return true;
+            if (ReferenceEquals(_instance, obj)) return true;
+
+            if (obj is FileRequestTransferWrapper<T> otherWrapper)
+            {
+                return Equals(_instance, otherWrapper._instance);
+            }
+
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return _instance != null ? _instance.GetHashCode() : 0;
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Used to call the CancelRequest function on the wrapped instance.
+        /// </summary>
+        /// <returns>
+        /// The result of canceling the wrapped request.
+        /// </returns>
+        public abstract Result CancelRequest();
+
+        /// <summary>
+        /// In a 1-to-1 wrapping, this method would simply call the "Release"
+        /// method on the wrapped class. However, the Dispose method is used
+        /// instead (and comments are included in the abstract DisposeInstance()
+        /// method that instruct implementing classes to call the Release()
+        /// method on the wrapped instance.
+        /// </summary>
+        public void Release()
+        {
+            Dispose();
+        }
+
+        /// <summary>
+        /// Used to neatly dispose of the wrapper and the instance that it
+        /// wraps.
+        /// </summary>
+        public void Dispose()
```

Comment:
Would it be a good idea to create a default deconstructor that also calls these?